readthedocs/search/serializers.py

```diff
@@ -0,0 +1,143 @@
+"""
+Serializers for the ES's search result object.
+
+.. note::
+   Some fields are re-named to make their meaning more clear.
+   They should be renamed in the ES index too.
+"""
+
+import itertools
+from operator import attrgetter
+
+from django.shortcuts import get_object_or_404
+from rest_framework import serializers
+
+from readthedocs.core.resolver import resolve
+from readthedocs.projects.models import Project
+
+
+class ProjectHighlightSerializer(serializers.Serializer):
+
+    name = serializers.ListField(child=serializers.CharField(), default=list)
+    slug = serializers.ListField(child=serializers.CharField(), default=list)
+    description = serializers.ListField(child=serializers.CharField(), default=list)
+
+
+class ProjectSearchSerializer(serializers.Serializer):
+
+    type = serializers.CharField(default='project', source=None, read_only=True)
+    name = serializers.CharField()
+    slug = serializers.CharField()
+    link = serializers.CharField(source='url')
+    highlight = ProjectHighlightSerializer(source='meta.highlight', default=dict)
+
+
+class PageHighlightSerializer(serializers.Serializer):
+
+    title = serializers.ListField(child=serializers.CharField(), default=list)
+
+
+class PageSearchSerializer(serializers.Serializer):
+
+    type = serializers.CharField(default='page', source=None, read_only=True)
+    project = serializers.CharField()
+    version = serializers.CharField()
+    title = serializers.CharField()
+    path = serializers.CharField(source='full_path')
+    link = serializers.SerializerMethodField()
+    highlight = PageHighlightSerializer(source='meta.highlight', default=dict)
+    inner_hits = serializers.SerializerMethodField()
+
+    def get_link(self, obj):
+        # TODO: optimize this to not query the db for each result.
```

Comment:
We are already doing one query per result, so this isn't something I'm introducing in this PR. We can fix this when merging the serializer from the api.

Reply:
It's doing more than 1 query if we're calling `resolve`.  We probably want to add this to the index, instead of computing it on render -- is that the plan?

Reply:
> We probably want to add this to the index

My plan was to so something similar to the subprojects, fetch the project once and resolve the main domain for that project and then pass it down here.